week-2/extended-project/valeria_and_friends/classes/top.rb

```diff
@@ -0,0 +1,85 @@
+require "http"
+require "json"
+
+class Top
+  attr_accessor :name_main_attr
+  attr_accessor :printable_data
+
+  BASE_URL = "https://swapi.co/api/"
+  TOP_NUMBER = 5
+  RELEVANT_DATA = "results"
+
+  def initialize (query, name_main_attr, relevant_keys)
+    @query = query
+    @name_main_attr = name_main_attr
+    @relevant_keys = relevant_keys
+    @printable_data = []
+  end
+
+  def give_me_the_top
+    json_answer = fetch_data(@query)
+    results = get_results(json_answer)
+    main_attributes = get_main_attribute(results, @name_main_attr)
+    top_ordered = get_top_of_results(main_attributes, results)
+    final_top = build_final_top(@relevant_keys, top_ordered)
+
+    @printable_data = final_top
+    final_top
+  end
+
+  def fetch_data(query)
+    @http = HTTP.get(BASE_URL + query + "/")
+    JSON.parse(@http)
+  end
+
+  # For local test
+  # def fetch_data(query)
+  #   data = File.open("./jsons/#{query}.json")
+  #   response = ""
+  #   data.each do |x|
+  #     response << x
+  #   end
+  #   data.close
+  #   JSON.parse(response)
+  # end
+
+  def get_results(json_complete)
+    json_complete[RELEVANT_DATA]
+  end
+
+  def get_main_attribute(result, main_attribute)
+    data_main_attributes = result.map { |item| item[main_attribute] }
+    parse_values_to_i(data_main_attributes)
+  end
+
+  def parse_values_to_i(arr_bidimensional)
+    arr_bidimensional.map! do |i|
+      (i == "unknown") ? 0 : i.to_i
+    end
+  end
+
+  def get_top_of_results(main_attributes, result)
+    top = []
+
+    TOP_NUMBER.times {
+      index_maxi = main_attributes.index(main_attributes.max)
+      top << result[index_maxi]
+      result.delete_at(index_maxi)
+      main_attributes.delete_at(index_maxi)
+    }
+    top
+  end
+
```

Comment:
The method "times" seems to me very interesting to use, but it would be more simple and short if your information store in a array so if you want get only a specify amount you would just need to use the index final to return the information.